internal/queue.go

```diff
@@ -0,0 +1,224 @@
+	return queue.client.ListenerService.registerListener(request, eventHandler,
+		func(registrationId *string) *ClientMessage {
+			return QueueRemoveListenerEncodeRequest(queue.name, registrationId)
+		}, func(clientMessage *ClientMessage) *string {
+			return QueueAddListenerDecodeResponse(clientMessage)()
+		})
+}
+
+func (queue *QueueProxy) Clear() (err error) {
+	request := QueueClearEncodeRequest(queue.name)
+	_, err = queue.Invoke(request)
+	return err
+}
+
+func (queue *QueueProxy) Contains(item interface{}) (found bool, err error) {
+	elementData, err := queue.validateAndSerialize(item)
+	if err != nil {
+		return false, err
+	}
+	request := QueueContainsEncodeRequest(queue.name, elementData)
+	responseMessage, err := queue.Invoke(request)
+	return queue.DecodeToBoolAndError(responseMessage, err, QueueContainsDecodeResponse)
+}
+
+func (queue *QueueProxy) ContainsAll(items []interface{}) (foundAll bool, err error) {
+	itemData, err := queue.validateAndSerializeSlice(items)
+	if err != nil {
+		return false, err
+	}
+	request := QueueContainsAllEncodeRequest(queue.name, itemData)
+	responseMessage, err := queue.Invoke(request)
+	return queue.DecodeToBoolAndError(responseMessage, err, QueueContainsAllDecodeResponse)
+}
+
+func (queue *QueueProxy) DrainTo(slice *[]interface{}) (movedAmount int32, err error) {
+	//TODO DrainTo should append to slice parameter, how will we do it?
+	var localSlice = *slice
+	request := QueueDrainToEncodeRequest(queue.name)
+	responseMessage, err := queue.Invoke(request)
+	resultSlice, err := queue.DecodeToInterfaceSliceAndError(responseMessage, err, QueueDrainToDecodeResponse)
+	if err != nil {
+		return 0, err
+	}
+	localSlice = append(localSlice, resultSlice)
+	slice = &localSlice
+	return int32(len(resultSlice)), nil
+}
+
+func (queue *QueueProxy) DrainToWithMaxSize(slice []interface{}, maxElements int32) (movedAmount int32, err error) {
+	request := QueueDrainToMaxSizeEncodeRequest(queue.name, maxElements)
+	responseMessage, err := queue.Invoke(request)
+	resultSlice, err := queue.DecodeToInterfaceSliceAndError(responseMessage, err, QueueDrainToMaxSizeDecodeResponse)
+	slice = append(slice, resultSlice)
+	return int32(len(slice)), err
+}
+
+func (queue *QueueProxy) IsEmpty() (empty bool, err error) {
+	request := QueueIsEmptyEncodeRequest(queue.name)
+	responseMessage, err := queue.Invoke(request)
+	return queue.DecodeToBoolAndError(responseMessage, err, QueueIsEmptyDecodeResponse)
+}
+
+func (queue *QueueProxy) Offer(item interface{}) (added bool, err error) {
+	itemData, err := queue.validateAndSerialize(item)
+	if err != nil {
+		return false, err
+	}
+	request := QueueOfferEncodeRequest(queue.name, itemData, 0)
+	responseMessage, err := queue.Invoke(request)
+	return queue.DecodeToBoolAndError(responseMessage, err, QueueOfferDecodeResponse)
+}
+
+func (queue *QueueProxy) OfferWithTimeout(item interface{}, timeoutInMilliSeconds int64) (added bool, err error) {
```

Comment:
We can offer a TimeUnit choice. See: https://github.com/hazelcast/hazelcast-go-client/blob/master/internal/map.go#L106

Reply:
That timeUnit exist in Java too. I'll keep it same as with Java interface and have only millis in queue otherwise we need to add these timeUnits in many more places.

Reply:
You say you will keep this as the same as Java. Java Client's Queue implementation has `TimeUnit` too: https://github.com/hazelcast/hazelcast/blob/master/hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientQueueProxy.java#L183

Reply:
You are right, all of them have. Let me update these.

Reply:
We can do it similar to `PutTransient` in map. https://github.com/hazelcast/hazelcast-go-client/blob/master/internal/map.go#L62